wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -305,7 +296,14 @@ public <T, E extends Exception> T execute(
       final Object[] jdbcMethodArgs)
       throws E {
 
-    final Connection conn = WrapperUtils.getConnectionFromSqlObject(methodInvokeOn);
+    final Connection conn;
+    if (AsynchronousMethodsHelper.ASYNCHRONOUS_METHODS.contains(methodName)) {
+      // The target driver may block on Statement.getConnection().
+      conn = this.pluginService.getCurrentConnection();
```

Comment:
I believe we need to get a connection from the Statement object and compare it to a current connection that is stored in PluginService.

Reply:
How target driver may block on Statement.getConnection()?